pkg/tsdb/influxdb/influxdb.go

```diff
@@ -96,17 +96,22 @@ func (e *InfluxDBExecutor) Query(ctx context.Context, dsInfo *models.DataSource,
 }
 
 func (e *InfluxDBExecutor) getQuery(dsInfo *models.DataSource, queries []*tsdb.Query, context *tsdb.TsdbQuery) (*Query, error) {
+	var (
+		query *Query
+		err   error
+	)
+
 	for _, v := range queries {
 
-		query, err := e.QueryParser.Parse(v.Model, dsInfo)
+		query, err = e.QueryParser.Parse(v.Model, dsInfo)
 		if err != nil {
 			return nil, err
 		}
-
-		return query, nil
+		goto found
 	}
-
 	return nil, fmt.Errorf("query request contains no queries")
+found:
```

Comment:
this seems like a very unconventional change for the worse (in terms of readability).  must be a better way?

Reply:
@torkelo yes, could you explain me this a little bit? I do not get why there is a for loop here.

Reply:
Good question, I think the model supports multiple queries but right now this is only used from alerting so we only needed to support batch executing 1 query at a time. So it just takes the first query.

Reply:
@torkelo ok. Should I remove that for loop and make a comment about this for future implementations?

Reply:
What is the problem with the loop? Is a length check index lookup better ?

Reply:
@torkelo this for loop is not needed, because it does only *one* loop, and as the warning say, it terminates unconditionally (no brake, continue, goto, etc). I will change this to parse the first element, and returns what corresponds to.

Reply:
@torkelo Done. PTAL.